main.py

```diff
@@ -150,6 +155,52 @@ def add_new_stream(self, stream_url=None, stream_quality=cfg[CONFIG_QUALITY]):
         # Also helps a lot with lag
         threading.Thread(target=self._add_new_stream, args=(stream_url, stream_quality)).start()
 
+    def add_new_scheduled_stream(self, *args, stream_url=None, stream_quality=cfg[CONFIG_QUALITY]):
+        """Add stream after certain delay"""
+        def schedule_stream():
+            self.save_stream_to_history(stream_url)
+            self._add_new_stream(stream_url, stream_quality)
+
+        """Schedules a new stream at given time"""
+        if not stream_url:
+            stream_url, ok1 = QtWidgets.QInputDialog.getText(
```

Comment:
If the stream_url changes here, the previously defined function `schedule_stream` will use the wrong stream_url.